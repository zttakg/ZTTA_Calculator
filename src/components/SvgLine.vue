
<template>
  <line
    :x1="x1"
    :y1="y1"
    :x2="x2"
    :y2="y2"
    :stroke-linecap="strokeLinecap"
    :stroke-width="strokeWidth"
    :stroke="strokeColor"
  />
</template>

<script>
export default {
  name: "SvgLine",

  props: {
    from: {
      type: Array,
      default() {
        return [20, 120];
      }
    },
    to: {
      type: Array,
      default() {
        return [120, 20];
      }
    },
    strokeWidth: {
      type: Number,
      default: 1
    },
    strokeLinecap: {
      type: String,
      default: "round"
    },
    strokeColor: {
      type: String,
      default: "#000"
    }
  },

  computed: {
    x1() {
      return this.from[0];
    },
    y1() {
      return this.from[1];
    },
    x2() {
      return this.to[0];
    },
    y2() {
      return this.to[1];
    }
  }
};
</script>
