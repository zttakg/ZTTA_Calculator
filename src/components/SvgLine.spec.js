import { shallowMount } from "@vue/test-utils";
import SvgLine from "./SvgLine";

describe("<SvgLine /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(SvgLine);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(SvgLine, {
      propsData: {
        from: [40, 400],
        to: [20, 200],
        strokeWidth: 5,
        strokeLinecap: "butt",
        strokeColor: "#f06"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
