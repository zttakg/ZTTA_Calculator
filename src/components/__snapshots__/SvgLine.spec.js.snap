// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgLine /> component should render all props 1`] = `<line x1="40" y1="400" x2="20" y2="200" stroke-linecap="butt" stroke-width="5" stroke="#f06"></line>`;

exports[`<SvgLine /> component should render correct without props 1`] = `<line x1="20" y1="120" x2="120" y2="20" stroke-linecap="round" stroke-width="1" stroke="#000"></line>`;
